.github/copilot-instructions.md

```diff
@@ -191,6 +191,29 @@ See [docs/UITests.md](../docs/UITests.md) for detailed UI testing setup and usag
 4. **Network Management**: Offline mode and better error handling
 5. **Performance**: Image caching and smooth scrolling optimizations
 
+### Development Best Practices
+
+**Documentation and Research Guidelines:**
+- **Always search Microsoft Learn (MS Learn)** when implementing .NET MAUI features, gestures, animations, or any platform-specific functionality - Microsoft Learn provides the most up-to-date and official guidance for .NET MAUI development
+- Reference official documentation before implementing new features to ensure best practices and optimal performance
+
+**MCP Server Configuration:**
+For enhanced Microsoft Learn integration, configure the Microsoft Docs MCP server:
+
+```json
+{
+  "mcpServers": {
+    "microsoft-docs-mcp": {
+      "type": "http",
+      "url": "https://learn.microsoft.com/api/mcp",
+      "tools": [ "microsoft_docs_search" ]
+    }
+  }
+}
+```
+
+This enables the `microsoft_docs_search` tool for direct access to Microsoft Learn documentation during development.
```

Comment:
No, we should mention to *use* the `microsoft_docs_search` tool for searching Microsoft docs.

We don't need to put the JSON snippet in here.